core/server/api_container/server/docker_compose_transpiler/docker_compose_transpiler.go

```diff
@@ -466,47 +471,55 @@ func getStarlarkFilesArtifacts(composeVolumes []types.ServiceVolumeConfig, servi
 		case types.VolumeTypeBind:
 			// Handle case where home path is reference
 			if strings.Contains(volume.Source, unixHomePathSymbol) {
-				return nil, map[string]string{}, stacktrace.NewError(
+				return nil, map[string]string{}, nil, stacktrace.NewError(
 					"Volume path '%v' uses '%v', likely referencing home path on a unix filesystem. Currently, Kurtosis does not support uploading from host filesystem. "+
 						"Place the contents of '%v' directory inside the package where the compose yaml exists and update the volume filepath to be a relative path",
 					volume.Source, unixHomePathSymbol, volume.Source)
 			}
 			// Handle case where upstream relative path is reference
 			if strings.Contains(volume.Source, upstreamRelativePathSymbol) {
-				return nil, map[string]string{}, stacktrace.NewError(
+				return nil, map[string]string{}, nil, stacktrace.NewError(
 					"Volume path '%v' uses '%v', likely referencing an upstream path on a filesystem. Currently, Kurtosis does not support uploading from host filesystem. "+
 						"Place the contents of '%v' directory inside the package where the compose yaml exists and update the volume filepath to be a relative path within the package.",
 					volume.Source, upstreamRelativePathSymbol, volume.Source)
 			}
 
 			// Assume that if an absolute path is specified, user wants to use volume as a persistence layer
 			// Additionally, assume relative paths are read-only
+			// GM - this should fail too as this could be outside the repository
 			shouldPersist = path.IsAbs(volume.Source)
```

Comment:
we are making the assumption that they want to use volume as persistence layer, so we create a persistent directory with `volume.Source` as the dir path on the service if they ref an absolute path on host machine - thus, we don't actually read anything from the filesystem. Maybe we can make this assumption/case more explicit

Reply:
Have removed my comment!